src/main/java/baritone/pathing/movement/movements/MovementParkour.java

```diff
@@ -104,17 +104,23 @@ public static void cost(CalculationContext context, int x, int y, int z, EnumFac
                 maxJump = 3;
             }
         }
+        
+        // check parkour jumps from smallest to largest for obstacles/walls and landing positions
+        int verifiedMaxJump = -1;
         for (int i = 2; i <= maxJump; i++) {
+        	verifiedMaxJump = i - 1;
```

Comment:
Then you can remove this line, no?

Reply:
no this sets verifiedMaxJump so that if it breaks it keeps the value

Reply:
(just realised the 2 comments are connected lol)

Reply:
To emulate what was there before verifiedMaxJump is set to `i-1` before the `break;` occurs, this way if it `break;`'s then it would keep the verifiedMaxJump value of `i-1` otherwise if it does not break then it exits out of the loop with `verifiedMaxJump = maxJump` effectively doing nothing.

Reply:
Right, I'm imagining that if you change the above line to `= 1`, and change the `verifiedMaxJump = maxJump` to `verifiedMaxJump = i` at the bottom of the loop, then by the time it hits a `break` verifiedMaxJump will already equal i-1. It will have been set by the previous time through the loop having run `verifiedMaxJump = i`, or if it's the first time through, by `verifiedMaxJump = 1`.

Reply:
this wouldn't work because there are cases where maxJump does not change at all exiting the loop while still having verifiedMaxJump the same as maxJump.

Reply:
And that is the same as the previous behavior seen here https://github.com/cabaletta/baritone/blob/2159773723c7b392a8ec2c67c57cca50e4ffb3d9/src/main/java/baritone/pathing/movement/movements/MovementParkour.java If the loop on 107 exits normally (no break), then the loop on 153 starts at normal maxJump. That would be the same, because the first loop goes up to maxJump, so the last thing it would do is set `verifiedMaxJump = i` but `i` equals `maxJump`.

Reply:
I applied what I think to that link https://github.com/cabaletta/baritone/blob/2159773723c7b392a8ec2c67c57cca50e4ffb3d9/src/main/java/baritone/pathing/movement/movements/MovementParkour.java

Look at: https://pastebin.com/KfiTNkrq

Reply:
The end `verifiedMaxJump = maxJump` could be encased in an if statement that checks if it is the last for loop iteration since that is the only time it is used

```
if (lastLoop) {
    verifiedMaxJump = maxJump;
} else {
    verifiedMaxJump = i;
}
```
Something like this? 

I see now... very smart.

`lastLoop = i == maxJump`

ok should work